account_invoice_purchase_quantity/tests/test_invoice_purchase_quantity.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2019 Camptocamp SA
+# Copyright ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo.tests.common import Form, TransactionCase
+
+
+class TestPurchaseOrderInvoicing(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+        cls.partner = cls.env["res.partner"].create({"name": "Test Partner"})
+        cls.product = cls.env["product.product"].create(
+            {
+                "name": "Product Test",
+                "type": "product",
+            }
+        )
+        cls._create_order()
+
+    @classmethod
+    def _create_order(cls):
+        with Form(cls.env["purchase.order"]) as order_form:
```

Comment:
I guess that `with Form()` is used to trigger some `onchange`. Otherwise I would advise not to use `with Form()`.

Reply:
> I guess that `with Form()` is used to trigger some `onchange`. Otherwise I would advise not to use `with Form()`.

Why ? It is designed for tests.(but for tests only)

Reply:
Just because they are explicitly calling `onchanges` which modifies the behavior that you would have if the use case could be called by a piece of code.
So if you test is not relying on `onchanges` it is safer not to use `Form`. Because in that case we presume that it can be called server side and that you will get the desired behavior. The idea being that if someone modifies your code and add an onchange to make it work, the test will fail... preserving the potentials server side calls.
Lastly to my understanding, `onchanges` should be changed into `computes`, except in some cases, so better not relying on them if not explicitly needed.

Reply:
This is not blocking though 🙂 I approved the PR and wrote a comment. Just to share it with you.